packages/builtin/_arithmetic.pony

```diff
@@ -0,0 +1,185 @@
+primitive _SignedArithmetic
+  fun fld[T: (SignedInteger[T, U] val & Signed), U: UnsignedInteger[U] val](x: T, y: T): T =>
+    if (y == T.from[U8](0)) or ((x == T.min_value()) and (y == T.from[I8](-1))) then
```

Comment:
i am not particularly fond of this implementation, but it is the only way i could come up with to get consistent behaviour in case of `y=0` or overflow. The implementation described in the RFC was much more elegant, but did not consider this edge case. 

Same for `mod` below.

If anyone has a good idea on how to get this to do less comparisons, please speak up.

Reply:
Is the edge case covered in the tests?

Reply:
These are the same checks that we do on every normal signed division, so I don't think this is a reason to hold up the PR.
For posterity, the checks on normal division are implemented as LLVM IR here:
https://github.com/ponylang/ponyc/blob/a62417003b081573bbb064a81146f6c3e34b6692/src/libponyc/codegen/genoperator.c#L139-L174

Reply:
@SeanTAllen tests for each edge case and each numeric type have been added. Github doesn't show them in the diff because it is a too large change.

- https://github.com/ponylang/ponyc/pull/2997/files#diff-79b0fcc2590ca809460bc4119931b946R1588
- https://github.com/ponylang/ponyc/pull/2997/files#diff-79b0fcc2590ca809460bc4119931b946R1622
- https://github.com/ponylang/ponyc/pull/2997/files#diff-79b0fcc2590ca809460bc4119931b946R1669
...

I also tested `NaN` and `Inf` behaviour for floats: https://github.com/ponylang/ponyc/pull/2997/files#diff-79b0fcc2590ca809460bc4119931b946R1694

@jemc that is relieving. So we are good here.

Reply:
Awesome.